QemuOpenBoardPkg: First version

theomacx86 opened the pull request.

theomacx86 committed:
QemuOpenBoardPkg: First version

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Dsc/Stage1.dsc.inc:
```diff
@@ -0,0 +1,76 @@
+
+  QemuOpenBoardPkg/Library/PlatformInitPei/PlatformInitPei.inf
+
+  
```
Two lines worth of whitespace (two spaces it seems), and no trailing newline

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Dsc/Stage2.dsc.inc:
```diff
@@ -0,0 +1,22 @@
+
+  [Components.X64]
+  MdeModulePkg/Universal/ResetSystemRuntimeDxe/ResetSystemRuntimeDxe.inf
+  MdeModulePkg/Bus/Pci/PciHostBridgeDxe/PciHostBridgeDxe.inf
```
No trailing newline

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Fdf/FlashMap.fdf.inc:
```diff
@@ -0,0 +1,63 @@
+SET gEfiMdeModulePkgTokenSpaceGuid.PcdFlashNvStorageVariableBase    = 0x0003C000
+SET gCpPlatFlashTokenSpaceGuid.PcdFlashFvNvStorageEventLogSize      = 0x00002000
+SET gEfiMdeModulePkgTokenSpaceGuid.PcdFlashNvStorageFtwWorkingSize  = 0x00002000
+SET gEfiMdeModulePkgTokenSpaceGuid.PcdFlashNvStorageFtwSpareSize    = gEfiMdeModulePkgTokenSpaceGuid.PcdFlashNvStorageVariableSize + gCpPlatFlashTokenSpaceGuid.PcdFlashFvNvStorageEventLogSize +gEfiMdeModulePkgTokenSpaceGuid.PcdFlashNvStorageFtwWorkingSize
```
\n

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Library/OpenQemuFwCfgLib.h:
```diff
@@ -0,0 +1,106 @@
+#ifndef QEMUOPENBOARDPKG_QEMUFWCFGLIB_H
```
no file header, no copyright

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Library/OpenQemuFwCfgLib.h:
```diff
@@ -0,0 +1,106 @@
+#define FW_CFG_ID         0x0001
+#define FW_CFG_FILE_DIR   0x0019
+
+
```
Talk about FwCfg a bit here. What is it, how does it work, yadda yadda

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Library/OpenQemuFwCfgLib.h:
```diff
@@ -0,0 +1,106 @@
+ *
+ * @param Size
+ * @param Buffer
+ * @return VOID
```
Remove the return VOID

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Library/QemuHardware.h:
```diff
@@ -0,0 +1,8 @@
+
+#define ICH9_DEVICE_ID   0x29C0
+
+#endif
```
\n

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Library/QemuHardware.h:
```diff
@@ -0,0 +1,8 @@
+#ifndef QEMUHW_H
```
Missing file + copyright header

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardBootManagerLib/BoardBootManager.c:
```diff
@@ -0,0 +1,105 @@
+/** @file
```
Is it impossible to use one of their BoardBootManagerLib? Or separate it into some sort of common library? Copying it straight up doesn't seem like a good idea to me

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.c:
```diff
@@ -0,0 +1,225 @@
+#include <Library/QemuHardware.h>
+#include <Library/HobLib.h>
+
+#define QEMU_IO_DEBUG_MAGIC  0xE9
```
Unused

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.c:
```diff
@@ -0,0 +1,225 @@
+
+  switch (DeviceID) {
+    case PIIX4_DEVICE_ID:
+      DEBUG ((DEBUG_INFO, "PIIX4\n"));
```
You should format this into a proper log message like "Board: PIIX4"

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.c:
```diff
@@ -0,0 +1,225 @@
+  This board service detects the board type.
+
+  @retval EFI_SUCCESS   The board was detected successfully.
+  @retval EFI_NOT_FOUND The board could not be detected.
```
EFI_NOT_FOUND seems like a weird status code for this

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.c:
```diff
@@ -0,0 +1,225 @@
+  VOID
+  )
+{
+  // Placeholder
```
Remove this placeholder stuff

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.c:
```diff
@@ -0,0 +1,225 @@
+  VOID
+  )
+{
+  DEBUG ((DEBUG_INFO, "BoardInitBeforeMemoryInit()\n"));
```
Same for this

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.c:
```diff
@@ -0,0 +1,225 @@
+  VOID
+  )
+{
+  DEBUG ((DEBUG_INFO, "BoardInitAfterMemoryInit()\n"));
```
And this

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.c:
```diff
@@ -0,0 +1,225 @@
+**/
+EFI_STATUS
+EFIAPI
+BoardInitBeforeTempRamExit (
```
And all of these

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.inf:
```diff
@@ -0,0 +1,29 @@
+  BoardInitLib.c
+
+[Packages]
+QemuOpenBoardPkg/QemuOpenBoardPkg.dec
```
Indentation here

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.inf:
```diff
@@ -0,0 +1,29 @@
+[Defines]
```
Missing file header here

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/BoardInitLib.inf:
```diff
@@ -0,0 +1,29 @@
+IoLib
+PciCf8Lib
+    
+[Ppis]
```
Useless sections - I'm fairly sure we don't need to specify them if they're empty

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/BoardInitLib/TestPointCheckLib.c:
```diff
@@ -0,0 +1,727 @@
+#include <Library/TestPointCheckLib.h>
```
Missing file header

Having to type all of this boilerplate code up is annoying :/

heatd replied:
Maybe we should actually verify some stuff?

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/OpenQemuFwCfgLib/OpenQemuFwCfgLib.c:
```diff
@@ -0,0 +1,128 @@
+/**
+  Checks for Qemu fw_cfg device by reading "QEMU" using the signature selector
+
+  @retval EFI_SUCCESS - The fw_cfg device is present
```
Discard the "-" and align phrases, like
```
@retval EFI_SUCCESS       The fw_cfg device is present
@retval EFI_UNSUPPORTED   The device is absent
```

heatd replied:
This applies to every other doxygen comment

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/OpenQemuFwCfgLib/OpenQemuFwCfgLib.c:
```diff
@@ -0,0 +1,128 @@
+  )
+{
+  EFI_STATUS  Status;
+  CHAR8       Control[4];
```
Control? Why is this named Control and not signature?

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/OpenQemuFwCfgLib/OpenQemuFwCfgLib.c:
```diff
@@ -0,0 +1,128 @@
+  QemuFwCfgSelectItem (FW_CFG_FILE_DIR);
+  QemuFwCfgReadBytes (sizeof (UINT32), &FilesCount);
+
+  FilesCount = SwapBytes32 (FilesCount);
```
Explain why you're swapping here in a comment

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PeiReportFvLib/PeiReportFvLib.c:
```diff
@@ -0,0 +1,286 @@
+#include <Guid/FirmwareFileSystem2.h>
+#include <Ppi/FirmwareVolumeInfo2.h>
+
+//Cast a FV pointer Pcd to an EFI_FIRMWARE_VOLUME_HEADER pointer
```
Nit: missing space

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PeiReportFvLib/PeiReportFvLib.c:
```diff
@@ -0,0 +1,286 @@
+#include <Ppi/FirmwareVolumeInfo2.h>
+
+//Cast a FV pointer Pcd to an EFI_FIRMWARE_VOLUME_HEADER pointer
+#define PCD_TO_FV_HEADER_PTR(Pcd)   ((EFI_FIRMWARE_VOLUME_HEADER *)(UINTN)PcdGet32 (Pcd))
```
I think we can do better than this.
Could we have a macro/function that did a lot of that work below?

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PeiReportFvLib/PeiReportFvLib.c:
```diff
@@ -0,0 +1,286 @@
+/** @file PeiReportFvLib.c
```
God cries every time firmware engineers copy a file :(

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Cpu.c:
```diff
@@ -0,0 +1,56 @@
+    return EFI_UNSUPPORTED;
+  }
+  
+  Status = QemuFwCfgSelectItem (QemuFwCfgItemSmpCpuCount);
```
Where is QemuFwCfgItemSmpCpuCount defined? I can't find it

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Cpu.c:
```diff
@@ -0,0 +1,56 @@
+#include <Library/HobLib.h>
+
+/**
+  Probe Qemu FW CFG device for current CPU count and report to MpInitLib
```
You're exposing internal details to the caller.
Try something like "Retrieve the number of CPUs present". Then inside the function, document why you're doing what you're doing.

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Cpu.c:
```diff
@@ -0,0 +1,56 @@
+  Status = QemuFwCfgIsPresent ();
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "QemuFwCfg not present, unable to detect CPU count \n"));
```
Nit: trailing space before the newline

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Pci.c:
```diff
@@ -0,0 +1,59 @@
+  PcdSet64S (PcdPciIoBase, PciIoBase);
+  PcdSet64S (PcdPciIoSize, PciIoSize);
+
+  PciMmio32Base = GetMemoryBelow4Gb ();
```
I think this is wrong. Shouldn't we find the hole instead of just counting memory? We risk shadowing DRAM with MMIO regions

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Pci.c:
```diff
@@ -0,0 +1,59 @@
+    return EFI_UNSUPPORTED;
+  }
+
+  DEBUG ((DEBUG_ERROR, "Memory below 4Gb: %x \n", PciMmio32Base));
```
trailing space before newline in the msg

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Memory.c:
```diff
@@ -0,0 +1,191 @@
+    DEBUG ((DEBUG_ERROR, "etc/e820 was not found \n"));
+    return EFI_UNSUPPORTED;
+  }
+
```
Explain the logic here for memory installation. Why are we doing what we are doing. Remember that most readers of your code lack the context to understand why things are done.

heatd replied:
and you will also slowly forget the context

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Memory.c:
```diff
@@ -0,0 +1,191 @@
+
+    if (ValidMemory) {
+      if ((E820Entry.Length > LargestE820Entry.Length) && (E820Entry.BaseAddr + E820Entry.Length <= SIZE_4GB)) {
+        DEBUG ((DEBUG_INFO, "New largest entry for PEI: BaseAddress %lx, Size %lx \n", LargestE820Entry.BaseAddr, LargestE820Entry.Length));
```
This seems too verbose to be kept, or at least kept as DEBUG_INFO

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Memory.c:
```diff
@@ -0,0 +1,191 @@
+/** @file
```
Something to think about for the future: most of this logic is x86 PC dependent. What happens if we add a QEMU arm64 board port? Or a new x86 board that doesn't have all of that VGA BIOS and FW shadowing stuff

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Pcie.c:
```diff
@@ -0,0 +1,90 @@
+    PciExBarBase.Uint32[0] | MCH_PCIEXBAR_BUS_FF | MCH_PCIEXBAR_EN
+    );
+
+  // Disable Pci MMIO above 4Gb
```
Why?

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/Pcie.c:
```diff
@@ -0,0 +1,90 @@
+  PcdSet64S (PcdPciMmio64Size, 0);
+
+  // Set Pci MMIO space below 4GB
+  PciBase = PcdGet64 (PcdPciExpressBaseAddress) + SIZE_256MB;
```
Side note: This logic seems to be reusable for PIIX4

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/PlatformInit.c:
```diff
@@ -0,0 +1,67 @@
+  Status = InstallMemory (PeiServices);
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "Memory installation failed\n"));
```
you can use %r and pass the status code to print the error messages associated with error codes

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/PlatformInit.c:
```diff
@@ -0,0 +1,67 @@
+    DEBUG ((DEBUG_INFO, "Memory installation success\n"));
+  }
+
+  // Report CPU core count to MPInitLib
```
Nit: MpInitLib

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/PlatformInit.c:
```diff
@@ -0,0 +1,67 @@
+
+  // Initialize PCI or PCIe based on current emulated system
+  if (DeviceId == INTEL_Q35_MCH_DEVICE_ID) {
+    DEBUG ((DEBUG_INFO, "Q35: Initialize PCIe\n"));
```
You should remove these DEBUG messages, they seem useless

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/PlatformInit.h:
```diff
@@ -0,0 +1,54 @@
+
+UINT32
+EFIAPI
+GetMemoryBelow4Gb(
```
Seems like it didn't get properly formatted

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/PlatformInit.h:
```diff
@@ -0,0 +1,54 @@
+#define Q35_PCI_IO_SIZE 0xA000
+
+#define PCI_MMIO_TOP_ADDRESS 0xFC000000
+
```
Add doxygen function comments for every single one of these

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/PlatformInit.h:
```diff
@@ -0,0 +1,54 @@
+EFIAPI
+MaxCpuInit(
+  VOID
+);
```
\n

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformSecLib/Ia32/SecEntry.nasm:
```diff
@@ -0,0 +1,113 @@
+    mov esp, ecx
+    add esp, edx
+
+    ; TODO: Multiprocessor support
```
Remove this TODO apparently. What are you going to do in regards to the BISTs?

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformSecLib/Ia32/SecEntry.nasm:
```diff
@@ -0,0 +1,113 @@
+    mov ecx, DWORD [ASM_PFX(PcdGet32 (PcdTemporaryRamBase))]
+    mov edx, DWORD [ASM_PFX(PcdGet32 (PcdTemporaryRamSize))]
+
+    ;Initialize the stack at the end of base + size
```
Nit: space here after the ";"

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformSecLib/Ia32/SecEntry.nasm:
```diff
@@ -0,0 +1,113 @@
+BITS 16
+align 4
+global ASM_PFX(_ModuleEntryPoint)
+ASM_PFX(_ModuleEntryPoint):
```
Explain what you're doing here in comments

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/PlatformSecLib/PlatformSecLib.c:
```diff
@@ -0,0 +1,140 @@
+#include <Library/LocalApicLib.h>
+
+EFI_PEI_CORE_FV_LOCATION_PPI  gEfiPeiCoreFvLocationPpi = {
+  (VOID *)0xFFE20000 // Could be done automatically at build
```
Could it? If so, we should do it

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/ResetSystemLib/ResetSystemLib.c:
```diff
@@ -0,0 +1,88 @@
+#include <Library/ResetSystemLib.h>
+
+/**
```
Implement please

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/ResetSystemLib/ResetSystemLib.c:
```diff
@@ -0,0 +1,88 @@
+#include <Library/ResetSystemLib.h>
```
Missing file header, copyright

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Library/ResetSystemLib/ResetSystemLib.inf:
```diff
@@ -0,0 +1,14 @@
+  ResetSystemLib.c
+
+[Packages]
+  MdeModulePkg/MdeModulePkg.dec
```
\n

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Policy/Library/PeiSiliconPolicyUpdateLib/PeiSiliconPolicyUpdateLib.c:
```diff
@@ -0,0 +1 @@
+#include "Library/SiliconPolicyUpdateLib.h"
```
\n

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Policy/Library/PeiSiliconPolicyUpdateLib/PeiSiliconPolicyUpdateLib.inf:
```diff
@@ -0,0 +1,28 @@
+
+[Protocols]
+
+[FeaturePcd]
```
Remove these empty sections

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/QemuOpenBoardPkg.dsc:
```diff
@@ -0,0 +1,112 @@
+[Defines]
```
Missing file header, copyright

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/QemuOpenBoardPkg.dec:
```diff
@@ -0,0 +1,32 @@
+[Defines]
```
Missing file header, copyright

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/QemuOpenBoardPkg.dsc:
```diff
@@ -0,0 +1,112 @@
+
+[PcdsFeatureFlag]
+  gMinPlatformPkgTokenSpaceGuid.PcdSmiHandlerProfileEnable|TRUE
+  gMinPlatformPkgTokenSpaceGuid.PcdUefiSecureBootEnable | FALSE
```
lack of consistency here in spacing between the pcd and the value

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/QemuOpenBoardPkg.dsc:
```diff
@@ -0,0 +1,112 @@
+[PcdsDynamicDefault]
+  gUefiOvmfPkgTokenSpaceGuid.PcdOvmfHostBridgePciDevId|0
+
+  #video setup
```
space after the # and a capital V

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/QemuOpenBoardPkg.fdf:
```diff
@@ -0,0 +1,239 @@
+!include QemuOpenBoardPkg/Include/Fdf/FlashMap.fdf.inc
```
File header, copyright

heatd commented on debug.log:
```diff
@@ -0,0 +1,456 @@
+build -a X64 -a IA32 -p Platform/Qemu/QemuOpenBoardPkg/QemuOpenBoardPkg.dsc -y Report.txt
```
Remove this

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Dsc/Stage1.dsc.inc:
```diff
@@ -0,0 +1,76 @@
+  #
+  # Enable stack and heap protections
+  #
+  #gEfiMdeModulePkgTokenSpaceGuid.PcdHeapGuardPropertyMask|0xFF
```
What are these commented out lines, why are they commented?

heatd commented on Silicon/Qemu/QemuOpenBoardPkg/Include/Dsc/Stage1.dsc.inc:
```diff
@@ -0,0 +1,76 @@
+
+[LibraryClasses]
+  PciSegmentInfoLib|MinPlatformPkg/Pci/Library/PciSegmentInfoLibSimple/PciSegmentInfoLibSimple.inf
+  #BoardInitLib|MinPlatformPkg/PlatformInit/Library/BoardInitLibNull/BoardInitLibNull.inf
```
Same as above

theomacx86 committed:
QemuOpenBoardPkg: Move to Platform, fix inconsistency
Add missing headers and descriptions to QemuFwCfgLib
Remove commented line in Stage 1 dsc

theomacx86 committed:
QemOpenBoardPkg: Remove duplicated code
Use IndustryStandard/E820
Use Ovmf's PIIX4 and Q35 definitions
Fix FDF whitespace inconsistencies

theomacx86 committed:
QemuOpenBoardPkg: Fix variable name, add missing headers
OpenQemuOpenBoardPkg variable names now respect edk2 coding style

theomacx86 committed:
QemuOpenBoardPkg: Enable SMM, remove unused code

heatd commented on Platform/Qemu/QemuOpenBoardPkg/Library/PlatformInitPei/PlatformInit.h:
```diff
@@ -52,3 +54,5 @@ EFIAPI
   VOID
 );
+
+#endif
```
\n

theomacx86 committed:
QemuOpenBoardPkg: Reorder FV definitions to match execution flow
FV Definition now matches execution flow and not the actual layout in FD

theomacx86 committed:
QemuOpenBoardPkg: Missing \n, remove double newlines

theomacx86 committed:
QemuOpenBoardPkg: Remove DXE components from stage 1

theomacx86 committed:
QemuOpenBoardPkg: Add SmmAccessPei for SMM support

theomacx86 committed:
QemuOpenBoardPkg: Reserve required memory for SMM

theomacx86 committed:
QemuOpenBoardPkg: Do only one check for SMRAM region
Instead of multiple checks, check for SMRAM overlap only once for each E820 entry

theomacx86 committed:
QemuOpenBoardPkg: Fix SMRAM memory stealing logic

heatd committed:
QemuOpenBoardPkg: Fix VS2019 build issues

Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>

theomacx86 committed:
QemuOpenBoardPkg: Relocate PlatformInitPei PEIM

iworam committed:
Clean up DEC file.
I already merged the missing MinPlatform PCD
We shouldn't need EventLog
We shouldn't use EFI or UEFI as a prefix for things that are not defined by UEFI specs.
Filename should be same line as @file I think.

iworam committed:
Clean up DEC file.
I already merged the missing MinPlatform PCD
We shouldn't need EventLog
We shouldn't use EFI or UEFI as a prefix for things that are not defined by UEFI specs.
Filename should be same line as @file I think.

heatd committed:
QemuOpenBoardPkg: Clean up memory detection

Check if we have enough memory to do our reservations, clean up DEBUG's,
fix GetMemoryBelow4Gb()'s return type.

Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>

theomacx86 committed:
Merge pull request #3 from theomacx86/pfalcato/QemuOpenBoardPkg-mmap

QemuOpenBoardPkg: Clean up memory detection

heatd committed:
QemuOpenBoardPkg: Fix the flash map

Now we're using a 8MiB flash area. This makes it more realistic, and
makes it so -drive if=pflash can load us.

Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>

heatd committed:
QemuOpenBoardPkg: Fix wrong DEBUG

Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>

heatd committed:
QemuOpenBoardPkg: Fix SMRAM reservation

Wrong type :|

Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>

iworam committed:
Fix FlashMap.fdf.inc

Match most of Pedro's changes
Move the FlashAreaSize/Base setting to QemuOpenBoardPkg.fdf since they have to be changed there in the FD...
Reserve areas in the FD for NvStorageVariable and FTW.  Before, you just had them all at 0x3C000.  This is badly broken as both variables and FTW are using the same physical base address in low memory.  I imagine reclaime would be terribly broken.  As would any OS use of that memory and variable writes.

iworam committed:
Fix FlashMap.fdf.inc

Match most of Pedro's changes
Move the FlashAreaSize/Base setting to QemuOpenBoardPkg.fdf since they have to be changed there in the FD...
Reserve areas in the FD for NvStorageVariable and FTW.  Before, you just had them all at 0x3C000.  This is badly broken as both variables and FTW are using the same physical base address in low memory.  I imagine reclaime would be terribly broken.  As would any OS use of that memory and variable writes.

iworam committed:
Minor coding style stuff.

Most people don't wrap spaces around |, though it is common to align vertically.
I think that your adding spaces is easier to read.  But mostly just be consistent within the file.

Updated some comments.

heatd committed:
QemuOpenBoardPkg: Fix SMM

We were using the wrong SmmCpuFeaturesLib and SmmCpuPlatformHookLib.

Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>

theomacx86 committed:
QemuOpenBoardPkg: Increase FVUEFIBOOT size
Increase FVUEFIBOOT size based on CLANG NOOPT build output

theomacx86 committed:
QemuOpenBoardPkg: Change SMM_ENABLED to SMM_REQUIRED, increase FVOSBOOT size

iworam committed:
FDF:
Increase commenting on NvStorage and FTW
Space out | operator.
Organize FV contents by stage and classification a bit better (debatable)
Add whitespace so groups of related drivers are more obviously related.

MD:
 Remove edk2/OvmfPkg from packages path as edk2 adds it.
 Minor additional commenting

DSC: Some whitespace cleanup (double indent).

heatd committed:
QemuOpenBoardPkg: Fix lower memory detection

Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>

heatd committed:
QemuOpenBoardPkg: Fix fdf and flashmap

Theo, please keep the size at 8MB, thanks.

Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>

heatd committed:
QemuOpenBoardPkg: Add SMBIOS support

Required by some OS installers.

Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>

theomacx86 committed:
QemuOpenBoardPkg: README V1, support PEI_ARCH and DXE_ARCH flags

theomacx86 committed:
Merge branch 'QemuOpenBoardPkgSquash' of github.com:theomacx86/edk2-platforms into QemuOpenBoardPkgSquash

theomacx86 committed:
QemuOpenBoardPkg: Incorrect link , missing trailing line in README

iworam committed:
Whitespace alignment.  I am not sold on this, but seems easier to read.
Put space before { in a few places missing.
Changed common to Common (camelcase style)
Moved library classes from individual driver overrides to standard use.
There should be a good reason to override a library class at the individual driver level.

Why is DXE using UP version of MpInitLib?  Switched to DxeMpInitLib.

Whitespace in fdf.inc

iworam committed:
BoardInitLib.c, why comment one is a placeholder

Add Missing file headers, delete dead code, indents, etc

TestPointCheckLib.c
You should implement copy simple read/write tests for the memory ranges since configuring those wrong is unintuitive for QEMU.
Why do you even have this anyway?  There are TestPointCheck libs for SEC, PEI, etc.  Seems like this can be deleted.

iworam committed:
Removed all the none/void @param and @return.  I checked MdePkg and it doesn't seem required.  The VOID in function prototype is still required.

iworam committed:
I don't understand the partial macro.  Macros suck for debug, but this is pretty basic so probably ok.

iworam committed:
Need to fix/resolve the ToDo
Whitespace cleanup
I don't think this lib works for X64.

iworam committed:
Do we really want the CpuDeadLoop us instead of ASSERTS?

Generally, people like putting error checking with function call block.  Not sure if required by style, but strong convention.

We should really distribute PlatformInitPeim into board support libs and silicon support if we are going to keep it.

theomacx86 committed:
QemuOpenBoardPkg: Remove SMM mention from MD, Compress FvUefiBoot

iworam committed:
edk2-platforms/Maintainers: Add QemuOpenBoardPkg

Add maintainers and reviewers for MinPlatform Arch QEMU board port.

Cc: Leif Lindholm <quic_llindhol@quicinc.com>
Cc: Michael D Kinney <michael.d.kinney@intel.com>
Cc: Isaac Oram <isaac.w.oram@intel.com>
Cc: Pedro Falcato <pedro.falcato@gmail.com>
Cc: Theo Jehl <theojehl76@gmail.com>
Signed-off-by: Isaac Oram <isaac.w.oram@intel.com>
